lms/djangoapps/discussion_api/forms.py

```diff
@@ -56,6 +59,19 @@ def clean_course_id(self):
         except InvalidKeyError:
             raise ValidationError("'{}' is not a valid course id".format(value))
 
+    def clean(self):
+        cleaned_data = super(ThreadListGetForm, self).clean()
+        exclusive_params_count = sum(
+            1 for param in self.EXCLUSIVE_PARAMS if cleaned_data.get(param)
+        )
+        if exclusive_params_count > 1:
+            raise ValidationError(
+                "The following query parameters are mutually exclusive: {}".format(
+                    ", ".join(self.EXCLUSIVE_PARAMS)
+                )
+            )
+        return cleaned_data
```

Comment:
I pondered this problem quite a bit. The alternatives I considered are:

1. Change the function to take a ThreadListGetForm.
2. Change the function to take a parameter dict and built the ThreadListGetForm itself.
3. Do not validate this in the form, and have the view catch the resulting ValueError and raise ValidationError.
4. Don't bother checking here and just exhibit unexpected behavior if the constraint is violated.

The main reason I don't like 1 and 2 is that part of the point of having the separate api module with these functions is to make it easier for other APIs to invoke this in-process, and either of those options would make doing that a bit more wonky. Thus, I erred on the side of making the implementation a little bit more wonky and the interface a little bit better. 3 just seems a little less semantically sound. I strongly dislike 4 for hopefully obvious reasons.